File: app/src/main/java/com/mona/moviesapp/popular_people/PopularDetailsScreen/model/ProfileImagesNetwork.java
package com.mona.moviesapp.popular_people.PopularDetailsScreen.model;

import android.os.AsyncTask;

import com.mona.moviesapp.popular_people.PopularDetailsScreen.controller.DetailsController;
import com.mona.moviesapp.popular_people.pojo.Profiles;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.MalformedURLException;
import java.net.URL;
import java.util.ArrayList;

public class ProfileImagesNetwork extends AsyncTask<String, String, String> {

    DetailsModel detailsModel;
    DetailsController detailsController;

    public ProfileImagesNetwork(DetailsModel detailsModel, DetailsController detailsController) {
        this.detailsModel = detailsModel;
        this.detailsController = detailsController;
    }

    @Override
    protected String doInBackground(String... urls) {

        HttpURLConnection httpURLConnection = null;
        BufferedReader reader = null;

        try {
            URL url = new URL(urls[0]);
            httpURLConnection = (HttpURLConnection) url.openConnection();
            httpURLConnection.connect();

            InputStream stream = httpURLConnection.getInputStream();
            reader = new BufferedReader(new InputStreamReader(stream));
            StringBuffer buffer = new StringBuffer();

            String line = "";
            while ((line = reader.readLine()) != null) {
                buffer.append(line);
            }
            return buffer.toString();
        } catch (MalformedURLException e) {
            e.printStackTrace();
        } catch (IOException e) {
            e.printStackTrace();
        }
        finally {
            if (httpURLConnection != null)
                httpURLConnection.disconnect();
            try {
                if (reader != null)
                    reader.close();
            } catch (IOException e) {
                e.printStackTrace();
            }
        }
        return null;
    }

    @Override
    protected void onPostExecute(String result) {
        super.onPostExecute(result);

        ArrayList<Profiles> profiles = new ArrayList<>();
        try {
            JSONObject popular_profiles = new JSONObject(result);
            JSONArray pop_profiles = popular_profiles.getJSONArray("profiles");
            for (int i = 0; i< pop_profiles.length(); i++) {
                JSONObject profileResult = pop_profiles.getJSONObject(i);
                Profiles pictures = new Profiles();
                pictures.setFile_path(profileResult.getString("file_path"));
                profiles.add(pictures);
            }

            detailsModel.detailsController.conGridRecycle(profiles);

        } catch (JSONException e) {
            e.printStackTrace();
        }
    }
}

File: app/src/main/java/com/mona/moviesapp/popular_people/PopularDetailsScreen/model/ProfileImagesNetwork.kt
package com.mona.moviesapp.popular_people.PopularDetailsScreen.model

import android.os.AsyncTask

import com.mona.moviesapp.popular_people.PopularDetailsScreen.controller.DetailsController
import com.mona.moviesapp.popular_people.pojo.Profiles

import org.json.JSONException
import org.json.JSONObject

import java.io.BufferedReader
import java.io.IOException
import java.io.InputStreamReader
import java.net.HttpURLConnection
import java.net.MalformedURLException
import java.net.URL

class ProfileImagesNetwork(internal var detailsModel: DetailsModel, internal var detailsController: DetailsController) : AsyncTask<String, String, String>() {

    override fun doInBackground(vararg urls: String): String? {

        var httpURLConnection: HttpURLConnection? = null
        var reader: BufferedReader? = null

        try {
            val url = URL(urls[0])
            httpURLConnection = url.openConnection() as HttpURLConnection
            httpURLConnection.connect()

            val stream = httpURLConnection.inputStream
            reader = BufferedReader(InputStreamReader(stream))
            val buffer = StringBuffer()

            do {
                var line = reader.readLine()
                buffer.append(line)
            }while (line != null)

            return buffer.toString()
        } catch (e: MalformedURLException) {
            e.printStackTrace()
        } catch (e: IOException) {
            e.printStackTrace()
        } finally {
            httpURLConnection?.disconnect()
            try {
                reader?.close()
            } catch (e: IOException) {
                e.printStackTrace()
            }

        }
        return null
    }

    override fun onPostExecute(result: String) {
        super.onPostExecute(result)

        try {
            val popular_profiles = JSONObject(result)
            val pop_profiles = popular_profiles.getJSONArray("profiles")
            for (i in 0 until pop_profiles.length()) {
                val profileResult = pop_profiles.getJSONObject(i)
                val pictures = Profiles()
                pictures.file_path = profileResult.getString("file_path")
                detailsModel.detailsController.getPopularDetails(pictures)
            }
            detailsModel.detailsController.conGridRecycle()

        } catch (e: JSONException) {
            e.printStackTrace()
        }

    }
}

